File: php/samples/SimpleOrm.Sample/src/Models/Tables/TransactionStatus.php
<?php

declare(strict_types=1);

namespace SimpleOrm\Sample\Models\Tables;

/** Stored as TEXT by case name, matched case-insensitively on read (§7.9); a pure enum, so the name is the value. */
enum TransactionStatus
{
    case Pending;
    case Completed;
    case Cancelled;
}

File: go/orm/sample/transaction_status.go
package sample

// TransactionStatus is stored as TEXT by name, matched case-insensitively on
// read (§7.9); a string-kinded enum, so the name is the value. Declaring
// EnumNames is what makes it an enum to the loader (orm.Enum).
type TransactionStatus string

const (
	Pending   TransactionStatus = "Pending"
	Completed TransactionStatus = "Completed"
	Cancelled TransactionStatus = "Cancelled"
)

// EnumNames lists the members in declaration order.
func (TransactionStatus) EnumNames() []string { return []string{"Pending", "Completed", "Cancelled"} }
